dbt/adapters/bigquery/impl.py

```diff
@@ -767,7 +767,7 @@ def get_table_options(
             # It doesn't apply the `require_partition_filter` option for a temporary table
             # so that we avoid the error by not specifying a partition with a temporary table
             # in the incremental model.
-            if config.get('require_partition_filter') is not None:
+            if config.get('require_partition_filter') and config.get('partition_by') is not None:
```

Comment:
I  think maybe we shall state out two notNone checks? 
```suggestion
            if config.get('require_partition_filter') is not None and config.get('partition_by') is not None:
```

Reply:
I think both should do the same, but I can definitely be more explicit if we need! Will wait for other opinions on this as well